File: python-solutions/1552-build-an-array-with-stack-operations/solution.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        vector<string> output;
        int target_idx = 0;

        for(int i = 1; i <= n; i++){
            output.push_back("Push");

            if( i == target[target_idx]){
                target_idx++;
            } else {
                output.push_back("Pop");
            }

            if (target_idx == target.size()){
                break;
            } 
        }       
    return output; 
    }     
};

File: python-solutions/1552-build-an-array-with-stack-operations/solution.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        stack_op = []
        target_idx = 0
        for i in range(1,n+1):
            stack_op.append("Push")
            if i == target[target_idx]:
                target_idx += 1
            else:
                stack_op.append("Pop")
            if target_idx == len(target):
                break
        return stack_op
